application/models/TemplateConfig.php

```diff
@@ -885,6 +885,74 @@ public static function uninstall($templatename)
         return false;
     }
 
+    /**
+     * Uninstalls all surveythemes that are being extended from the supplied surveytheme name
+     * @param $templateName
+     * @return void
+     * @throws CDbException
+     */
+    public static function uninstallThemesRecursive($templateName): void
+    {
+        $extendedTemplates = Template::model()->findAll('extends=:templateName', [':templateName' => $templateName]);
+        if (!empty($extendedTemplates)) {
+            foreach ($extendedTemplates as $extendedTemplate) {
+                self::uninstallThemesRecursive($extendedTemplate->name);
+            }
+        }
+        self::uninstall($templateName);
+    }
+
+    /**
+     * Checks if a theme is valid
+     * Can be extended with more checks in the future if needed
+     * @param $themeName
+     * @param $themePath
+     * @param bool $redirect
+     * @return bool
+     * @throws CDbException
+     */
+    public static function validateTheme($themeName, $themePath, bool $redirect = true): bool
+    {
+        // check compatability with current limesurvey version
+        $isCompatible = self::isCompatible($themePath);
+        if (!$isCompatible) {
+            self::uninstallThemesRecursive($themeName);
+            if ($redirect) {
+                App()->setFlashMessage(
+                    sprintf(
+                        gT("Theme '%s' has been uninstalled because it's not compatible with this LimeSurvey version."),
+                        $themeName
+                    ),
+                    'error'
+                );
+                App()->getController()->redirect(["themeOptions/index", "#" => "surveythemes"]);
+                App()->end();
+            }
+        }
+        // add more tests here
+
+        // all checks succeeded, continue loading the theme
+        return true;
+    }
+
+    /**
+     * Checks if theme is compatible with the current limesurvey version
+     * @param $themePath
+     * @param bool $redirect
+     * @return bool
+     */
+    public static function isCompatible($themePath): bool
```

Comment:
I feel like such a function should not be static? `$surveyTheme->isCompatible()` would be more natural or expected OOP design, no?

Reply:
Ah wait, that already exists.

Reply:
Or perhaps `$surveyTheme->config->isCompatible()`

Reply:
its a bit problematic since there are functionscalling this who are also static and dont have a tempalte class